File: android/core/ui/src/main/kotlin/com/dualmusic/core/ui/components/CreditPill.kt
package com.dualmusic.core.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.dualmusic.core.ui.theme.DualMusicTheme
import java.text.NumberFormat
import java.util.Locale

/**
 * Pastille affichant un solde/prix en crédits, avec l'accent rose néon.
 *
 * ```
 * CreditPill(credits = 1250.0)              // « 💎 1 250 »
 * CreditPill(credits = 50.0, label = "Vote")
 * ```
 */
@Composable
fun CreditPill(
    credits: Double,
    modifier: Modifier = Modifier,
    label: String? = null,
) {
    val colors = DualMusicTheme.colors
    Row(
        modifier = modifier
            .clip(RoundedCornerShape(DualMusicTheme.radii.pill))
            .background(colors.accent)
            .padding(horizontal = DualMusicTheme.spacing.md, vertical = DualMusicTheme.spacing.sm),
    ) {
        Text("💎 ", color = colors.accentForeground)
        Text(
            text = formatCredits(credits),
            color = colors.accentForeground,
            fontWeight = FontWeight.SemiBold,
        )
        if (label != null) {
            Text("  $label", color = colors.accentForeground)
        }
    }
}

/** Formatage avec séparateur de milliers, sans décimales inutiles. */
private fun formatCredits(credits: Double): String {
    val nf = NumberFormat.getNumberInstance(Locale.getDefault())
    nf.maximumFractionDigits = if (credits % 1.0 == 0.0) 0 else 2
    return nf.format(credits)
}

File: ios/Packages/CoreUI/Sources/CoreUI/Components/CreditPill.swift
import SwiftUI

/// Pastille affichant un solde/prix en crédits, avec l'accent rose néon.
///
/// Utilisée dans la barre de wallet, les panneaux de cadeaux, les prix de tickets.
/// ```swift
/// CreditPill(credits: 1250)          // « 💎 1 250 »
/// CreditPill(credits: 50, label: "Vote")
/// ```
public struct CreditPill: View {
    @Environment(\.dmTheme) private var theme

    private let credits: Double
    private let label: String?

    public init(credits: Double, label: String? = nil) {
        self.credits = credits
        self.label = label
    }

    public var body: some View {
        HStack(spacing: 6) {
            Text("💎")
            Text(formatted).font(DMFont.mono)
            if let label { Text(label).font(DMFont.caption).opacity(0.8) }
        }
        .foregroundStyle(theme.colors.accentForeground)
        .padding(.horizontal, theme.spacing.md)
        .padding(.vertical, theme.spacing.sm)
        .background(theme.colors.accent)
        .clipShape(RoundedRectangle(cornerRadius: theme.radius.pill, style: .continuous))
    }

    /// Formatage avec séparateur de milliers (espace insécable), sans décimales inutiles.
    private var formatted: String {
        let f = NumberFormatter()
        f.numberStyle = .decimal
        f.maximumFractionDigits = credits.truncatingRemainder(dividingBy: 1) == 0 ? 0 : 2
        return f.string(from: NSNumber(value: credits)) ?? "\(credits)"
    }
}

#Preview {
    HStack {
        CreditPill(credits: 1250)
        CreditPill(credits: 50, label: "Vote")
    }
    .padding()
    .background(DMColors.dark.background)
    .dualMusicTheme(.dark)
}
